[Bug] Collection created in 2.15 can't be opened in 2.14.6

###### Reproduction Steps

1. Create collection in 2.15
2. Create Note
3. Open in 2.14.6
4. Review


###### Expected Result

Works

###### Actual Result

```
2021-05-25 20:33:47.416 9612-9612/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ichi2.anki.e, PID: 9612
    com.ichi2.utils.JSONException: org.json.JSONException: No value for req
        at com.ichi2.utils.JSONObject.get(JSONObject.java:2)
        at org.json.JSONObject.getJSONArray(JSONObject.java:595)
        at com.ichi2.utils.JSONObject.getJSONArray(JSONObject.java:2)
        at com.ichi2.libanki.Models.emptyCard(Models.java:3)
        at com.ichi2.libanki.Card.isEmpty(Card.java:1)
        at com.ichi2.anki.AbstractFlashcardViewer.displayString(AbstractFlashcardViewer.java:1)
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:6)
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:1)
        at com.ichi2.anki.Reviewer.displayCardQuestion(Reviewer.java:2)
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.displayNext(AbstractFlashcardViewer.java:10)
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.onProgressUpdate(AbstractFlashcardViewer.java:1)
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:4)
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:1)
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:791)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7660)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: org.json.JSONException: No value for req
        at org.json.JSONObject.get(JSONObject.java:400)
        at com.ichi2.utils.JSONObject.get(JSONObject.java:1)
        at org.json.JSONObject.getJSONArray(JSONObject.java:595) 
        at com.ichi2.utils.JSONObject.getJSONArray(JSONObject.java:2) 
        at com.ichi2.libanki.Models.emptyCard(Models.java:3) 
        at com.ichi2.libanki.Card.isEmpty(Card.java:1) 
        at com.ichi2.anki.AbstractFlashcardViewer.displayString(AbstractFlashcardViewer.java:1) 
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:6) 
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:1) 
        at com.ichi2.anki.Reviewer.displayCardQuestion(Reviewer.java:2) 
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.displayNext(AbstractFlashcardViewer.java:10) 
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.onProgressUpdate(AbstractFlashcardViewer.java:1) 
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:4) 
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:1) 
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:791) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7660) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
2021-05-25 20:33:47.418 9612-9612/? E/ACRA: ACRA caught a JSONException for com.ichi2.anki.e
    com.ichi2.utils.JSONException: org.json.JSONException: No value for req
        at com.ichi2.utils.JSONObject.get(JSONObject.java:2)
        at org.json.JSONObject.getJSONArray(JSONObject.java:595)
        at com.ichi2.utils.JSONObject.getJSONArray(JSONObject.java:2)
        at com.ichi2.libanki.Models.emptyCard(Models.java:3)
        at com.ichi2.libanki.Card.isEmpty(Card.java:1)
        at com.ichi2.anki.AbstractFlashcardViewer.displayString(AbstractFlashcardViewer.java:1)
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:6)
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:1)
        at com.ichi2.anki.Reviewer.displayCardQuestion(Reviewer.java:2)
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.displayNext(AbstractFlashcardViewer.java:10)
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.onProgressUpdate(AbstractFlashcardViewer.java:1)
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:4)
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:1)
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:791)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7660)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: org.json.JSONException: No value for req
        at org.json.JSONObject.get(JSONObject.java:400)
        at com.ichi2.utils.JSONObject.get(JSONObject.java:1)
        at org.json.JSONObject.getJSONArray(JSONObject.java:595) 
        at com.ichi2.utils.JSONObject.getJSONArray(JSONObject.java:2) 
        at com.ichi2.libanki.Models.emptyCard(Models.java:3) 
        at com.ichi2.libanki.Card.isEmpty(Card.java:1) 
        at com.ichi2.anki.AbstractFlashcardViewer.displayString(AbstractFlashcardViewer.java:1) 
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:6) 
        at com.ichi2.anki.AbstractFlashcardViewer.displayCardQuestion(AbstractFlashcardViewer.java:1) 
        at com.ichi2.anki.Reviewer.displayCardQuestion(Reviewer.java:2) 
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.displayNext(AbstractFlashcardViewer.java:10) 
        at com.ichi2.anki.AbstractFlashcardViewer$NextCardHandler.onProgressUpdate(AbstractFlashcardViewer.java:1) 
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:4) 
        at com.ichi2.async.CollectionTask.onProgressUpdate(CollectionTask.java:1) 
        at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:791) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7660) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/Models.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/Models.java
@@ -201,6 +201,13 @@ public class Models {
             m.put("mod", mCol.getTime().intTime());
             m.put("usn", mCol.usn());
             // TODO: fix empty id problem on _updaterequired (needed for model adding)
+            if (!isModelNew(m)) {
+                // this fills in the `req` chunk of the model. Not used on AnkiDroid 2.15+ or AnkiDesktop 2.1.x
+                // Included only for backwards compatibility (to AnkiDroid <2.14 etc)
+                // https://forums.ankiweb.net/t/is-req-still-used-or-present/9977
+                // https://github.com/ankidroid/Anki-Android/issues/8945
+                _updateRequired(m);
+            }
             if (templates) {
                 _syncTemplates(m);
             }
@@ -957,6 +964,25 @@ public class Models {
      * Required field/text cache
      * ***********************************************************************************************
      */
+
+    private void _updateRequired(Model m) {
+        if (m.isCloze()) {
+            // nothing to do
+            return;
+        }
+        JSONArray req = new JSONArray();
+        List<String> flds = m.getFieldsNames();
+        JSONArray templates = m.getJSONArray("tmpls");
+        for (JSONObject t: templates.jsonObjectIterable()) {
+            Object[] ret = _reqForTemplate(m, flds, t);
+            JSONArray r = new JSONArray();
+            r.put(t.getInt("ord"));
+            r.put(ret[0]);
+            r.put(ret[1]);
+            req.put(r);
+        }
+        m.put("req", req);
+    }
 
     @SuppressWarnings("PMD.UnusedLocalVariable") // 'String f' is unused upstream as well
     private Object[] _reqForTemplate(Model m, List<String> flds, JSONObject t) {

--- a/AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.java
+++ b/AnkiDroid/src/test/java/com/ichi2/libanki/ModelTest.java
@@ -466,6 +466,48 @@ public class ModelTest extends RobolectricTest {
     }
 
     @Test
+    public void test_req() {
+
+        Collection col = getCol();
+        Models mm = col.getModels();
+        Model basic = mm.byName("Basic");
+        assertTrue(basic.has("req"));
+        reqSize(basic);
+        JSONArray r = basic.getJSONArray("req").getJSONArray(0);
+        assertEquals(0, r.getInt(0));
+        assertTrue(Arrays.asList(new String[] {REQ_ANY, REQ_ALL}).contains(r.getString(1)));
+        assertEquals(1, r.getJSONArray(2).length());
+        assertEquals(0, r.getJSONArray(2).getInt(0));
+
+        Model opt = mm.byName("Basic (optional reversed card)");
+        reqSize(opt);
+
+        r = opt.getJSONArray("req").getJSONArray(0);
+        assertTrue(Arrays.asList(new String[] {REQ_ANY, REQ_ALL}).contains(r.getString(1)));
+        assertEquals(1, r.getJSONArray(2).length());
+        assertEquals(0, r.getJSONArray(2).getInt(0));
+
+
+        assertEquals(new JSONArray("[1,\"all\",[1,2]]"), opt.getJSONArray("req").getJSONArray(1));
+
+        // testing any
+        opt.getJSONArray("tmpls").getJSONObject(1).put("qfmt", "{{Back}}{{Add Reverse}}");
+        mm.save(opt, true);
+        assertEquals(new JSONArray("[1, \"any\", [1, 2]]"), opt.getJSONArray("req").getJSONArray(1));
+        // testing null
+        opt.getJSONArray("tmpls").getJSONObject(1).put("qfmt", "{{^Add Reverse}}{{/Add Reverse}}");
+        mm.save(opt, true);
+        assertEquals(new JSONArray("[1, \"none\", []]"), opt.getJSONArray("req").getJSONArray(1));
+
+        opt = mm.byName("Basic (type in the answer)");
+        reqSize(opt);
+        r = opt.getJSONArray("req").getJSONArray(0);
+        assertTrue(Arrays.asList(new String[] {REQ_ANY, REQ_ALL}).contains(r.getString(1)));
+        // TODO: Port anki@4e33775ed4346ef136ece6ef5efec5ba46057c6b
+        assertEquals(new JSONArray("[0]"), r.getJSONArray(2));
+    }
+
+    @Test
     @Config(qualifiers = "en")
     public void regression_test_pipe() {
         Collection col = getCol();

